integreat_cms/cms/models/contact/contact.py

```diff
@@ -125,6 +127,47 @@ def search(cls, region: Region, query: str) -> QuerySet:
             .distinct()
         )
 
+    @classmethod
+    def suggest(cls, **kwargs: Any) -> list[dict[str, Any]]:
+        r"""
+        Suggests keywords for contact search
+
+        :param \**kwargs: The supplied kwargs
+        :return: Json object containing all matching elements, of shape {title: str, url: str, type: str}
+        """
+        results: list[dict[str, Any]] = []
+
+        region = kwargs["region"]
+        query = kwargs["query"]
+        archived_flag = kwargs["archived_flag"]
+        language_slug = kwargs["language_slug"]
+
+        results.extend(
+            {
+                "title": contact.name,
+                "url": None,
+                "type": "contact",
+            }
+            for contact in cls.objects.filter(
+                name__icontains=query,
+                location__region=region,
+                archived=archived_flag,
+            ).distinct()
+        )
+        if language_slug:
+            results.extend(
+                {
+                    "title": poi_translation.title,
+                    "url": None,
+                    "type": "contact",
+                }
+                for poi_translation in POITranslation.search(
+                    region, language_slug, query
+                ).filter(poi__archived=False)
+            )
+
+        return results
```

Comment:
This doesn't seem done yet, you copied it over from POITranslation and forgot to otherwise adopt it. Though I don't know where it is even used at, and if at all –

While trying to look this up I noticed that I committed a crime, because people looking through `urls/protected.py` will never notice that `utils.search_content_ajax` and `utils.search_contact_ajax` are different. I'm not quite sure I remember why a separate implementation was necessary, but now we ended up with two similar but different implementations of doing a search. And maybe `search_content_ajax` for contacts isn't used anywhere.

Reply:
We need take POITranslations into account because contacts are seachable not only by `name` but also by `location`.


`search_contact_ajax` is being used when user try to insert a contact card. 
`search_content_ajax` is a keyword suggestion in the contact list (`object_type == "contact"` in this case).

But I found those two very confusing. We should probably rename them. Added as a new ToDo in #3558